### PR TITLE
Go samples created for sending first request, SSL backend server and OAuth backend server

### PR DESCRIPTION
**Pull request by maria-mcparland**

### CODE REVIEW

File: go/backend-example/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Go Backend Samples
+
+In this folder there are two code samples for connecting to our APIs using Go.
```
Comment (jondey) [inline go/backend-example/README.md:3]: Not sure, maybe turn it around? 
There are two code samples in this folder...

File: go/backend-example/oauth/proxy.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"sample_authentication_code/pkg/samples"
+)
+
+func main() {
+	remote, err := url.Parse("https://api-mock.payments.jpmorgan.com")
+	if err != nil {
+		panic(err)
+	}
+	access_token_url := "https://id.payments.jpmorgan.com/am/oauth2/alpha/access_token"
+	client_id := "3988f709-a8f9-47c3-9d28-c9fcc466d5c0"
```
Comment (jondey) [inline go/backend-example/oauth/proxy.go:30]: note to keep these secret somewhere?

File: go/backend-example/oauth/proxy.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	client_secret := "y_OAgjpmVcEL1vpPDI9SXjIkdUYWBXpl8UIWHIVIlRW45zsl2q0EjfL_ffGkUDn02X2_RDZWe1buriJPrTwmeA"
+	//digital_key := ""
+	handler := func(p *httputil.ReverseProxy) func(http.ResponseWriter, *http.Request) {
+		return func(w http.ResponseWriter, r *http.Request) {
+			log.Println(r.URL)
+			r.Host = remote.Host
+			// Access token generation. We are using the code from our sample authentication folder
+			accessToken, err := samples.GetAccessToken(access_token_url, client_id, client_secret)
+			if err != nil {
+				log.Printf("Error getting access token: %v", err)
```
Comment (jondey) [inline go/backend-example/oauth/proxy.go:40]: Code should fail here is no token?

File: go/backend-example/oauth/proxy.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+			}
+			w.Header().Set("Authorization", "Bearer "+accessToken)
+			// JWT/Digital Signature generation. This is required for some of our POST requests.
+			//modifyRequestBody(w, r, digital_key)
```
Comment (jondey) [inline go/backend-example/oauth/proxy.go:44]: Comment required to uncomment the line if using digital sigs?

Comment (jondey) [inline go/backend-example/oauth/proxy.go:44]: could run the method if digital_key is not empty/nil etc?

then just need a comment above the digital_key variable to say comment in and add this if your are using ?

can we pass those values in with command line as env vars? then can use os env etc?

File: go/backend-example/ssl/proxy.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+	// Register the handler and start the server
+	http.HandleFunc("/", handler(proxy))
+	err = http.ListenAndServe(":8001", nil)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func modifyRequestBody(w http.ResponseWriter, r *http.Request, digital_key string) {
```
Comment (jondey) [inline go/backend-example/ssl/proxy.go:74]: worth refactoring this block of code into its own class?

File: go/sample-authentication-code/pkg/samples/getJWT.go [hunk 6]
```diff
@@ -10,10 +10,12 @@ import (
 	"github.com/golang-jwt/jwt/v5"
 )
 
-func generate_digital_signature(digital_key string, body map[string]interface{}) string {
+func GenerateDigitalSignature(digital_key string, body map[string]interface{}) string {
 	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims(body))
 	signKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(digital_key))
-
+	if err != nil {
+		fmt.Println(err)
```
Comment (jondey) [inline go/sample-authentication-code/pkg/samples/getJWT.go:17]: should be handling errors?

File: go/backend-example/oauth/proxy.go (hunk 4 above)
Comment (maria-mcparland) [reply go/backend-example/oauth/proxy.go]: good idea, updated

**Review by jondey — APPROVED** [review]
Approved

**Pull request merged**